lib/client/keystore.go

```diff
@@ -669,7 +694,13 @@ func matchesWildcard(hostname, pattern string) bool {
 }
 
 // GetKnownHostKeys returns all known public keys from `known_hosts`.
-func (fs *fsLocalNonSessionKeyStore) GetKnownHostKeys(hostname string) ([]ssh.PublicKey, error) {
+func (fs *fsLocalNonSessionKeyStore) GetKnownHostKeys(hostname string) (keys []ssh.PublicKey, retErr error) {
+	unlock, err := acquireFileLock(fs.knownHostsPath(), 5*time.Second)
```

Comment:
Should this function use the read lock?

Reply:
I don't really think for this use case it's worth it to complicate matter and introduce write/read locks distinction. Here it's a very rare edge case when this lock would matter and latency difference with using just one type of lock should be negligible. Of couse if it was more performance sensitive path it would be different story, but it's not the case here. 
What do you think?

Reply:
We may have situations where even a single tsh command is doing things in parallel. But agreed even in those cases the performance improvement is probably negligible. 

My personal reason to use an RLock here is it indicates this function is not modifying the file. I got that confused when I first read this.